plugins/modules/keycloak_authentication.py

```diff
@@ -359,8 +359,10 @@ def main():
     module = AnsibleModule(argument_spec=argument_spec,
                            supports_check_mode=True,
                            required_one_of=([['token', 'auth_realm', 'auth_username', 'auth_password']]),
-                           required_together=([['auth_realm', 'auth_username', 'auth_password']])
-                           )
+                           required_together=([['auth_realm', 'auth_username', 'auth_password']]),
+                           required_by={
+                               'refresh_token': 'auth_realm',
+                           })
```

Comment:
A bit of a pain, but being helpful to those coming after us:
```suggestion
                           required_together=([['auth_realm', 'auth_username', 'auth_password']]),
                           required_by={'refresh_token': 'auth_realm'},
                           )
```
We get leaner PRs with this, like one for plugins/modules/keycloak_clientscope_type.py below

Reply:
No problem, done.